### PR TITLE
MAYA-112530: Fix group command return object

### PR DESCRIPTION
**Pull request by HamedSabri-adsk**

### CODE REVIEW

File: lib/mayaUsd/ufe/ProxyShapeHierarchy.h [hunk 1]
```diff
@@ -66,7 +66,7 @@ class MAYAUSD_CORE_PUBLIC ProxyShapeHierarchy : public Ufe::Hierarchy
 #ifdef UFE_V2_FEATURES_AVAILABLE
     Ufe::SceneItem::Ptr
     createGroup(const Ufe::Selection& selection, const Ufe::PathComponent& name) const override;
-    Ufe::UndoableCommand::Ptr
+    Ufe::InsertChildCommand::Ptr
```
Comment (HamedSabri-adsk) [inline lib/mayaUsd/ufe/ProxyShapeHierarchy.h:69]: adopt to new UFE interface change.

File: lib/mayaUsd/ufe/UsdUndoCreateGroupCommand.h [hunk 2]
```diff
@@ -26,7 +26,7 @@ namespace MAYAUSD_NS_DEF {
 namespace ufe {
 
 //! \brief UsdUndoCreateGroupCommand
-class MAYAUSD_CORE_PUBLIC UsdUndoCreateGroupCommand : public Ufe::CompositeUndoableCommand
+class MAYAUSD_CORE_PUBLIC UsdUndoCreateGroupCommand : public Ufe::InsertChildCommand
```
Comment (HamedSabri-adsk) [inline lib/mayaUsd/ufe/UsdUndoCreateGroupCommand.h:29]: UsdUndoCreateGroupCommand  now drives from InsertChildCommand and its insertedChild() should return the group sceneItem. 

This class also no longer need to drive from CompositeUndoableCommand and instead _groupCompositeCmd handles the command composition. 

This also makes it possible to have a clean undo()/redo() calls.

File: test/lib/ufe/testUngroupCmd.py [hunk 3]
```diff
@@ -110,13 +110,13 @@ def testUngroupUndoRedo(self):
...
-        cmds.group(ufe.PathString.string(sphere1Path), 
-                   ufe.PathString.string(sphere2Path))
+        groupName = cmds.group(ufe.PathString.string(sphere1Path), 
+                               ufe.PathString.string(sphere2Path))
 
         # verify selected item is "group1"
         self.assertEqual(len(self.globalSn), 1)
         groupItem = self.globalSn.front()
-        self.assertEqual(groupItem.nodeName(), "group1")
+        self.assertEqual(groupItem.nodeName(), groupName)
```
Comment (HamedSabri-adsk) [inline test/lib/ufe/testUngroupCmd.py:119]: Now, I can rely on the group name :)

File: lib/mayaUsd/ufe/UsdUndoCreateGroupCommand.h [hunk 4]
```diff
@@ -60,7 +60,7 @@ class MAYAUSD_CORE_PUBLIC UsdUndoCreateGroupCommand : public Ufe::InsertChildCom
 private:
     UsdSceneItem::Ptr  _parentItem;
     Ufe::PathComponent _name;
-    UsdSceneItem::Ptr  _group;
+    UsdSceneItem::Ptr  _groupItem;
```
Comment (HamedSabri-adsk) [inline lib/mayaUsd/ufe/UsdUndoCreateGroupCommand.h:63]: _group is very generic.

File: lib/mayaUsd/ufe/ProxyShapeHierarchy.cpp [hunk 5]
```diff
@@ -226,7 +226,11 @@ Ufe::SceneItem::Ptr ProxyShapeHierarchy::createGroup(
     return createdItem;
 }
 
+#ifdef UFE_V3_FEATURES_AVAILABLE
 Ufe::InsertChildCommand::Ptr ProxyShapeHierarchy::createGroupCmd(
+#elif
+Ufe::UndoableCommand::Ptr ProxyShapeHierarchy::createGroupCmd(
+#endif
```
Comment (HamedSabri-adsk) [inline lib/mayaUsd/ufe/ProxyShapeHierarchy.cpp:233]: Adding UFE guards around the return object for createGroupCmd

**Review by seando-adsk — APPROVED** [review]

File: lib/mayaUsd/ufe/ProxyShapeHierarchy.cpp [hunk 6]
```diff
@@ -220,19 +220,19 @@ Ufe::SceneItem::Ptr ProxyShapeHierarchy::createGroup(
         = UsdUndoCreateGroupCommand::create(usdItem, selection, name.string());
     if (cmd) {
         cmd->execute();
-#ifdef UFE_V3_FEATURES_AVAILABLE
+#if (UFE_PREVIEW_VERSION_NUM >= 3001)
```
Comment (HamedSabri-adsk) [inline lib/mayaUsd/ufe/ProxyShapeHierarchy.cpp:223]: @seando-adsk  Adjusted some of the ifdefs to adopt to 0.3.1 changes.

**Review by seando-adsk — APPROVED** [review]

**Comment by HamedSabri-adsk** [comment]
@kxl-adsk marking this PR as ready to merge.  2022_python2_linux failure is not relevant.

https://github.com/Autodesk/maya-usd/actions/runs/1015060515

**Pull request merged**